src/FuncSharp/Numeric/DigitExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace FuncSharp;
+
+public static class DigitExtensions
+{
+    public static IOption<Digit> AsDigit(this char value)
+    {
+        return Digit.Create(value);
+    }
+
+    public static IEnumerable<Digit> FilterDigits(this string value)
+    {
+        if (value is null)
+        {
+            return Enumerable.Empty<Digit>();
+        }
+        return value.Select(Digit.Create).Flatten();
```

Comment:
btw I really hate the naming: `Flatten` intuitively means "flatten nested arrays" and not "skip empty options"

Reply:
Just replace `arrays` with `container` or `box` and it works. 
 You can flatten 
 - `Array<Array<T>>`
 - `IEnumerable<IEnumerable<T>>`
 - `Task<Task<T>>`
 - `IOption<IOption<T>>`
 - `Try<Try<T>>`
 
 so then when you thing about it, why wouldn't you be able to flatten 
 - `Array<IEnumerable<T>>`
 
 And if that can be done, then why not
 - `IEnumerable<IOption<T>>`
 - `IOption<IEnumerable<T>>`
 
 I could imaging flattening `IEnumerable<T?>`. Of course we just called the method `ExceptNulls` but it's also a flatten.